src/core/amount.js

```diff
@@ -105,12 +106,12 @@ Amount.NaN = function() {
   return result;                      // but let's be careful
 };
 
-Amount.from_components_unsafe = function(value: Value, currency: Currency,
+Amount.from_components_unsafe = function(value: Value, currency: string,
   issuer: string, isNative: boolean
 ) {
   const result = new Amount(value);
   result._is_native = isNative;
-  result._currency = currency;
+  result._currency = normalizeCurrency(currency);
```

Comment:
I think we should remove `normalizeCurrency` from here - it is caller's responsibility to provide correct values.